### PR TITLE
[WIP][Discussion] Ensure the global batch size is not changed even the horovod size changes.

### PR DESCRIPTION
**Pull request by brightcoder01**
Please use the following command the do the verification.

```SHELL
horovodrun -np 3 -H localhost:3 python model_zoo/mnist/mnist_train_tfv1_fixed_global_size.py
```

The variable `backward_passes_per_step` of the first two workers is 3 and the value of the 3rd worker is 2.

### CODE REVIEW

File: model_zoo/mnist/customized_optimizers.py [hunk 1]
```diff
@@ -0,0 +1,376 @@
+import horovod.tensorflow as hvd
+import tensorflow as tf 
+
+from tensorflow.python.training import optimizer
+
+
+class UpdateVariableBySizeHook(tf.train.SessionRunHook):
+    def __init__(self, updatable_tensor, total_count):
```
Comment (workingloong) [inline model_zoo/mnist/customized_optimizers.py]: Add a docstring to describe the `total_count` and `updatable_tensor`.

File: model_zoo/mnist/customized_optimizers.py [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+            name = "Distributed{}".format(type(optimizer).__name__)
+        super(ElasticDistributedOptimizer, self).__init__(name=name, use_locking=use_locking)
+
+        self._optimizer = optimizer
+        self._allreduce_grads = hvd._make_allreduce_grads_fn(
+            name, device_dense, device_sparse, compression, sparse_as_dense, op,
+            gradient_predivide_factor)
+
+        self._agg_helper = None
+        if backward_passes_per_step > 1:
```
Comment (workingloong) [inline model_zoo/mnist/customized_optimizers.py]: In the elastic training, the `backward_passes_per_step` need to change. So, should we use the `agg_helper` defaultly?

Comment (brightcoder01) [reply model_zoo/mnist/customized_optimizers.py]: Good idea, updated.

File: model_zoo/mnist/customized_optimizers.py (hunk 1 above)
Comment (brightcoder01) [reply model_zoo/mnist/customized_optimizers.py]: Good idea. Docstring is added and do some renaming for readability.

**Comment by brightcoder01** [comment]
The official solution is submitted in PR #2409 2409

**Pull request closed without merging**